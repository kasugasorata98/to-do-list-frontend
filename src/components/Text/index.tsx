import React, { CSSProperties, ReactNode } from "react";
import { Text as ChakraText } from "@chakra-ui/react";
import { colors } from "@/styles/colors";

const Text: React.FC<{
  style?: CSSProperties | undefined;
  children?: ReactNode;
}> = ({ style, children }) => {
  return (
    <ChakraText
      style={{
        color: colors.textColor,
        ...style,
      }}
    >
      {children}
    </ChakraText>
  );
};

export default Text;
