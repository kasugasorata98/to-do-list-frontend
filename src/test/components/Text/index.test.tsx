import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Text from "@/components/Text";

describe("Text component", () => {
  it("matches snapshot", () => {
    const { container } = render(<Text>Hello world</Text>);
    expect(container).toMatchSnapshot();
  });
  it("renders text correctly", () => {
    const text = "Hello, world!";
    const { getByText } = render(<Text>{text}</Text>);
    const renderedText = getByText(text);
    expect(renderedText).toBeInTheDocument();
  });

  it("applies style prop to text element", () => {
    const style = { fontWeight: "bold" };
    const { getByText } = render(<Text style={style}>Text with style</Text>);
    const textElement = getByText("Text with style");
    expect(textElement).toHaveStyle("font-weight: bold");
  });
});
