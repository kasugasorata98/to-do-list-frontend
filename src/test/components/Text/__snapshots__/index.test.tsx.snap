// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component matches snapshot 1`] = `
<div>
  <p
    class="chakra-text css-0"
    style="color: rgb(226, 234, 241);"
  >
    Hello world
  </p>
</div>
`;
